lib/cli/run-helpers.js

```diff
@@ -107,29 +107,49 @@ exports.handleRequires = async (requires = [], {ignoredPlugins = []} = {}) => {
 };
 
 /**
- * Collect and load test files, then run mocha instance.
+ * Runs Mocha once in serial mode.
+ *
+ * 1. Finds test files
+ * 2. Runs global setup fixtures, if any
+ * 3. Loads test files
+ * 4. Runs tests
+ * 5. Runs global teardown fixtures, if any
  * @param {Mocha} mocha - Mocha instance
- * @param {Options} [opts] - Command line options
- * @param {boolean} [opts.exit] - Whether or not to force-exit after tests are complete
+ * @param {Options} opts - Command line options
  * @param {Object} fileCollectParams - Parameters that control test
  *   file collection. See `lib/cli/collect-files.js`.
  * @returns {Promise<Runner>}
  * @private
  */
-const singleRun = async (mocha, {exit}, fileCollectParams) => {
+const singleRun = async (mocha, options, fileCollectParams) => {
   const files = collectFiles(fileCollectParams);
   debug('single run with %d file(s)', files.length);
   mocha.files = files;
 
+  let context = {};
+  if (mocha.globalSetupEnabled() && mocha.hasGlobalSetupFixtures()) {
+    context = await mocha.runGlobalSetup(context);
+  }
   // handles ESM modules
   await mocha.loadFilesAsync();
-  return mocha.run(exit ? exitMocha : exitMochaLater);
+
+  const done = options.exit ? exitMocha : exitMochaLater;
+
+  return mocha.run(async (...args) => {
+    if (mocha.globalTeardownEnabled() && mocha.hasGlobalTeardownFixtures()) {
+      await mocha.runGlobalTeardown(context);
+    }
+    done(...args);
+  });
```

Comment:
Aside: https://github.com/mochajs/mocha/pull/4771/files#r1511173111 also talks about refactoring these shared wrappers. If I were to end up taking ownership over this PR then I'd probably apply that refactor. Food for thought.